Backend/editor/tests/transform_tests.go

```diff
@@ -0,0 +1,136 @@
+package editor
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTransformPoint(t *testing.T) {
+	var pos1 = []int{1, 2, 3}
+	var pos2 = []int{1, 2, 4}
+	assert.Equal(t, transformPoint(pos1, pos2), 2)
+
+	var pos3 = []int{1, 0}
+	var pos4 = []int{1, 0, 3, 2}
+	assert.Equal(t, transformPoint(pos3, pos4), 1)
+}
+
+func TestEffectIndependence(t *testing.T) {
+	// Effect independence condition 1 (refer to https://arxiv.org/pdf/1512.05949.pdf)
+	var pos1 = []int{2, 2, 3, 4}
+	var pos2 = []int{1, 2, 4, 4}
+	assert.Equal(t, effectIndependent(pos1, pos2, transformPoint(pos1, pos2)), true)
```

Comment:
either export these functions in transform.go by capitalising them (not ideal) or move your transform_test.go file to the same directory as transform.do